src/utilities/hooks/useContract.js

```diff
@@ -0,0 +1,312 @@
+          to: web3.config.contracts.Planter.address,
+          data: joinPlanterData,
+          value: 0,
+          operation: OperationType.Call,
+        },
+      ]
+
+      const nextNonce = await safeApiKit.getNextNonce(safeAddress)
+
+      const safeTransaction = await safe.createTransaction({
+        safeTransactionData,
+        options: { nonce: nextNonce },
+      })
+
+      const senderAddress = await safeOwner.getAddress()
+
+      const safeTxHash = await safe.getTransactionHash(safeTransaction)
+
+      const signature = await safe.signTransactionHash(safeTxHash)
+
+      // Propose transaction to the service
+      await safeApiKit.proposeTransaction({
+        safeAddress: await safe.getAddress(),
+        safeTransactionData: safeTransaction.data,
+        safeTxHash,
+        senderAddress,
+        senderSignature: signature.data,
+      })
+      setContractResponse({
+        hash: { detail: safeTxHash, message: 'Transaction submitted successfully.' },
+        success: null,
+        error: null,
+      })
+    } catch (error) {
+      setContractResponse({
+        hash: null,
+        success: null,
+        error: { detail: error, message: error.message },
+      })
+    }
+  }
+
+  // Grant high Level Role //
+  const handleGrantHighLevel = async (role, address) => {
+    let roleHash = null
+    if (role === 'verifier') {
+      roleHash = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('VERIFIER_ROLE'))
+    } else if (role === 'admin') {
+      roleHash = '0x0000000000000000000000000000000000000000000000000000000000000000'
+    } else if (role === 'datamanager') {
+      roleHash = ethers.utils.keccak256(ethers.utils.toUtf8Bytes('DATA_MANAGER_ROLE'))
+    }
+
+    try {
+      const config = await prepareWriteContract({
+        address: web3.config.contracts.AR.address,
+        abi: web3.config.contracts.AR.abi,
+        functionName: 'grantRole',
+        args: [roleHash, address],
+      })
+      const { hash } = await writeContract(config)
+      setContractResponse({
+        hash: null,
+        success: { detail: 'Planter Submitted.', message: 'Role submitted successfully.' },
+        error: null,
+      })
+      const tx = await waitForTransaction({
+        hash: hash,
+      })
+      setContractResponse({
+        hash: { detail: tx, message: 'Transaction submitted successfully.' },
+        success: null,
+        error: null,
+      })
+    } catch (error) {
+      setContractResponse({
+        hash: null,
+        success: null,
+        error: { detail: error, message: error.details },
+      })
+    }
+  }
+
+  const prepareData = (data) => {
+    const contryCode = findCountryCode(data.user.mobileCountry)
+    const lat = Math.trunc(data.application.latitude * Math.pow(10, 6))
+    const lng = Math.trunc(data.application.longitude * Math.pow(10, 6))
+    const reff = data.application.Referrer
+      ? data.application.Referrer
+      : '0x0000000000000000000000000000000000000000'
```

Comment:
You can create a function to generate this type of empty hashes
There is some other usage at the above of this file
A function to prefix `0x` and append zero by passing a param
For example:

```ts
export function emptyHash(zeroCount: number) {
  return `0x${new Array(zeroCount).fill('0').join('')}`
}
```